agent/internal/command.go

```diff
@@ -0,0 +1,80 @@
+package internal
+
+import (
+	"github.com/sqreen/go-agent/agent/internal/backend/api"
+	"github.com/sqreen/go-agent/agent/internal/plog"
+)
+
+type CommandManager struct {
+	logger   *plog.Logger
+	agent    CommandManagerAgent
+	handlers map[string]CommandHandler
+}
+
+type CommandHandler func() api.CommandResult
+
+// CommandManagerAgent defines the expected agent SDK and allows to easily
+// implement functional tests by mocking it up.
+type CommandManagerAgent interface {
+	InstrumentationEnable() error
+	InstrumentationDisable() error
+}
+
+func NewCommandManager(agent CommandManagerAgent, logger *plog.Logger) *CommandManager {
+	mng := &CommandManager{
+		agent:  agent,
+		logger: plog.NewLogger("command", logger),
+	}
+
+	// Note: using Go's reflection to call methods would be slower.
+	mng.handlers = map[string]CommandHandler{
```

Comment:
Can this be defined outsied of the func as a const?

Reply:
I didn't because here I need the pointer to the agent which is now dynamically created (cf. the previous PR your reviewed where I removed every global ^^).

Of course, I could pass the pointer to the agent at run-time when calling the handler, but then this map would be statically defined (ie. data initialized by the runtime during the program' startup), and would therefore involve memory overhead even when the agent is disabled.

As a general rule from now, I will make global only what is const and likely perfectly optimized by the compiler, such as number constants and strings.

Reply:
LGTM